core/src/main/java/com/oath/oak/KeysManager.java

```diff
@@ -34,19 +33,15 @@ public int length() {
     }
 
     public void writeKey(K key, int ki) {
-        assert keys.position() == 0;
         ByteBuffer byteBuffer = keys.duplicate();
-        byteBuffer.position(ki);
-        byteBuffer.limit(ki + keySerializer.calculateSize(key));
-        byteBuffer = byteBuffer.slice();
-        assert byteBuffer.position() == 0;
+        byteBuffer.position(keys.position() + ki);
+        byteBuffer.limit(keys.position() + ki + keySerializer.calculateSize(key));
```

Comment:
Same here:
```suggestion
        byteBuffer.limit(byteBuffer.position() + keySerializer.calculateSize(key));
```

Also, not sure if for similar reasons we should use `keySerializer.calculateSize(`**`byteBuffer`**`))`, or if it will misbehave with new position/limit